yarn-project/noir-compiler/src/compile/noir/dependencies/github-dependency-resolver.ts

```diff
@@ -0,0 +1,133 @@
+import { createDebugOnlyLogger } from '@aztec/foundation/log';
+
+import { join, sep } from 'node:path';
+import { unzip } from 'unzipit';
+
+import { FileManager } from '../file-manager/file-manager.js';
+import { NoirDependencyConfig, NoirGitDependencyConfig } from '../package-config.js';
+import { NoirPackage } from '../package.js';
+import { DependencyResolver } from './dependency-resolver.js';
+
+/**
+ * Downloads dependencies from github
+ */
+export class GithubDependencyResolver implements DependencyResolver {
+  #fm: FileManager;
+  #log = createDebugOnlyLogger('');
+
+  constructor(fm: FileManager) {
+    this.#fm = fm;
+  }
+
+  /**
+   * Resolves a dependency from github. Returns null if URL is for a different website.
+   * @param _pkg - The package to resolve the dependency for
+   * @param dependency - The dependency configuration
+   * @returns asd
+   */
+  async resolveDependency(_pkg: NoirPackage, dependency: NoirDependencyConfig): Promise<NoirPackage | null> {
+    // TODO accept ssh urls?
+    // TODO github authentication?
+    if (!('git' in dependency) || !dependency.git.startsWith('https://github.com')) {
+      return null;
+    }
+
+    const archivePath = await this.#fetchZipFromGithub(dependency);
+    const libPath = await this.#extractZip(dependency, archivePath);
+    return NoirPackage.open(libPath, this.#fm);
+  }
+
+  async #fetchZipFromGithub(dependency: Pick<NoirGitDependencyConfig, 'git' | 'tag'>): Promise<string> {
+    if (!dependency.git.startsWith('https://github.com')) {
+      throw new Error('Only github dependencies are supported');
+    }
+
+    const url = resolveGithubCodeArchive(dependency, 'zip');
+    const localArchivePath = join(
+      'archives',
+      url.pathname
+        // remove leading slash
+        .slice(1)
+        .replaceAll(sep, '_'),
+    );
+
+    // TODO should check signature before accepting any file
+    if (this.#fm.hasFileSync(localArchivePath)) {
+      this.#log('using cached archive', { url: url.href, path: localArchivePath });
+      return localArchivePath;
+    }
+
+    const response = await fetch(url, {
+      method: 'GET',
+    });
+
+    if (!response.ok || !response.body) {
+      throw new Error(`Failed to fetch ${url}: ${response.statusText}`);
+    }
+
+    await this.#fm.writeFile(localArchivePath, response.body);
+    return localArchivePath;
+  }
+
+  async #extractZip(dependency: NoirGitDependencyConfig, archivePath: string): Promise<string> {
+    const gitUrl = new URL(dependency.git);
+    const extractLocation = join(
+      'libs',
+      gitUrl.pathname
+        // remove leading slash
+        .slice(1)
+        .replaceAll('/', '_') +
```

Comment:
```suggestion
        .replaceAll(sep, '_') +
```
Though I'm pretty sure nothing we're doing works on Windows, but oh well